### PR TITLE
variety bug fix

### PR DESCRIPTION
**Pull request by PaleNeutron**
## Proposed Changes

  - `PyQuery.find` returns `HtmlElement` not `PyQuery`
  - see https://github.com/imWildCat/scylla/compare/main...PaleNeutron:main#diff-57108e489d5c7bc7e48c7c4da76657e55a3269abc284db3a0aabd750a1eddb7bR25, instantiate provider in worker, pass class instead of object to avoid `UnpickleableError: Cannot pickle <type 'thread.lock'> objects`
  - other provider fix

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
 RUN apt-get install -y libssl-dev
+WORKDIR /root
 
-RUN mkdir -p /var/www/scylla
-WORKDIR /var/www/scylla
+COPY . .
+COPY --from=node-build /root/scylla/assets ./scylla
 
-RUN pip install scylla
+RUN pip install -r requirements.txt -i https://mirrors.aliyun.com/pypi/simple/  && python setup.py install
```
Comment (imWildCat) [inline Dockerfile:16]: We don't need this because GH Actions are running overseas.

I will find a way to fix the CI.

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
-RUN pip install scylla
+COPY --from=node-build /root/scylla/assets ./scylla/assets
+COPY requirements.txt .
+RUN pip install -r requirements.txt -i https://mirrors.aliyun.com/pypi/simple/
+RUN python -m playwright install
+COPY . .
+RUN python setup.py install
 
-FROM python:3.9-slim
+FROM python:3.9-slim as prod
```
Comment (imWildCat) [inline Dockerfile:21]: Why `as prod` is needed

File: Dockerfile (hunk 1 above)
Comment (imWildCat) [reply Dockerfile]: The CI has been fixed.

File: Dockerfile (hunk 2 above)
Comment (PaleNeutron) [reply Dockerfile:21]: I added a  docker-compose file which is under testing, `prod` is used for specific target in docker-compose file's `build`

Comment (PaleNeutron) [reply Dockerfile:21]: oops, changes for testing are sending by pr 😭

Comment (PaleNeutron) [reply Dockerfile:21]: please just review b29969249ac4df1cc178aa019f33d23cd9b17d30 😭😭😭😭

Comment (imWildCat) [reply Dockerfile:21]: Can you remove this change?

File: scylla/scheduler.py [hunk 3]
```diff
@@ -111,6 +111,7 @@ def feed_from_db():
 class Scheduler(object):
 
     def __init__(self):
+        self.manager = Manager()
```
Comment (imWildCat) [inline scylla/scheduler.py:114]: Could you explain what's the purpose of this change?

**Review by imWildCat — CHANGES_REQUESTED** [review]

File: scylla/scheduler.py (hunk 3 above)
Comment (PaleNeutron) [reply scylla/scheduler.py:114]: A legacy change. When I face `UnpickleableError: Cannot pickle <type 'thread.lock'> objects`, I thought it is due to `multiprocessing.Queue` object can not be picked.

But I finally found the problem is caused by initialize provider in the main process.

It could be removed.

File: Dockerfile (hunk 2 above)
Comment (PaleNeutron) [reply Dockerfile:21]: Yes, of course, give one night and I'll merge them into another pr.

File: scylla/scheduler.py (hunk 3 above)
Comment (imWildCat) [reply scylla/scheduler.py:114]: Thanks for the explanation!

File: Dockerfile (hunk 2 above)
Comment (imWildCat) [reply Dockerfile:21]: Sure! Let me know if your PR is ready. You could also mark this one as a draft PR if needed.

**Review by imWildCat — CHANGES_REQUESTED** [review]

**Comment by PaleNeutron** [comment]
Hi, commits are neat now.

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+parts/
+sdist/
+var/
+wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
```
Comment (imWildCat) [inline .dockerignore:30]: We're not using `PyInstaller`. Could you remove it?

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
+
+# Translations
+*.mo
```
Comment (imWildCat) [inline .dockerignore:53]: ditto, we're not using these either

File: Dockerfile [hunk 6]
```diff
@@ -1,23 +1,33 @@
-FROM python:3.9-slim as build
+FROM node:lts-buster as node-build
```
Comment (imWildCat) [inline Dockerfile:1]: maybe just `node:lts`?

File: Dockerfile [hunk 7]
```diff
@@ -1,23 +1,33 @@
...
+WORKDIR /root
+COPY package.json .
+RUN npm install
+COPY . .
+RUN make assets-build
 
-RUN apt-get update && apt-get install -y g++ gcc libxslt-dev make libcurl4-openssl-dev build-essential
 
+FROM python:3.9-slim as build
```
Comment (imWildCat) [inline Dockerfile:9]: maybe `python-build` for clarity?

File: Dockerfile [hunk 8]
```diff
@@ -1,23 +1,33 @@
...
+RUN apt-get install -y libgbm-dev gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget
 
-COPY --from=build /usr/local/lib/python3.6/site-packages/ /usr/local/lib/python3.6/site-packages/
+COPY --from=build /usr/local/lib/python3.9/site-packages/ /usr/local/lib/python3.9/site-packages/
+COPY --from=build /root/.cache/ms-playwright /root/.cache/ms-playwright
```
Comment (imWildCat) [inline Dockerfile:30]: Nice work!

File: scylla/providers/proxy_list_provider.py [hunk 9]
```diff
@@ -35,14 +35,14 @@ def parse(self, document: PyQuery) -> [ProxyIP]:
     def urls(self) -> [str]:
         ret = []
         first_url = 'http://proxy-list.org/english/index.php?p=1'
-        sub = first_url[0:first_url.rfind('/')] # http://proxy-list.org/english
+        # sub = first_url[0:first_url.rfind('/')] # http://proxy-list.org/english
```
Comment (imWildCat) [inline scylla/providers/proxy_list_provider.py:38]: Since of commenting out of this line, could we just remove it since we have history?

**Review by imWildCat — CHANGES_REQUESTED** [review]

File: .dockerignore (hunk 4 above)
Comment (PaleNeutron) [reply .dockerignore:30]: Since `.dockerignore` is just a copy of `.gitignore ` I do not review it🤣 

So, maybe we could remove it from both files?

File: Dockerfile (hunk 6 above)
Comment (PaleNeutron) [reply Dockerfile:1]: `node:lts` equals `node:stretch-lts`, maybe a little old and not smaller than buster version.

BTW, I choose `buster` not `slim-buster` because node compile need python environment.

File: Dockerfile (hunk 7 above)
Comment (PaleNeutron) [reply Dockerfile:9]: Good idea

File: .dockerignore (hunk 4 above)
Comment (imWildCat) [reply .dockerignore:30]: Not a blocker for this PR. We can clean it up later.

File: Dockerfile (hunk 6 above)
Comment (imWildCat) [reply Dockerfile:1]: Thanks! Let's use `buster` then

**Review by imWildCat — APPROVED** [review]
Thanks for the effort!

I really appreciate it!

**Comment by imWildCat** [comment]
@PaleNeutron I'm merging this one.
